src/pipeline.ml

```diff
@@ -243,5 +277,98 @@ let v ?channel ~ocluster () =
      ("base", Current_docker.push_manifest ?auth:Conf.auth ~tag:(Tag.v distro) opam_images |> Current.ignore_value)
       :: ocaml_images
       @ archive_images)
-  )
+
+let windows_distro_pipeline ~ocluster repos distro_label distro_versions =
+  let distro_pipeline multiarches distro =
+    let opam_images, ocaml_images, archive_image =
+      let arch_results =
+        let arches = Conf.arches_for ~distro in
+        List.map (Arch.pipeline ~ocluster ~repos ~distro) arches in
+      List.fold_left (fun (aa,ba,ca) (a,b,c) ->
+        let ca = match ca,c with Some v, _ -> Some v | None, v -> v in
+        a::aa, b::ba, ca) ([], [], None) arch_results in
+    let (multiarch_images, ocaml_images) =
+      all_switches ocaml_images |> List.filter_map @@ begin fun switch ->
+        let images = List.filter_map (Switch_map.find_opt switch) ocaml_images in
+        if images = [] then None
+        else (
+          let full_tag = Tag.v distro ~switch in
+          let tags =
+            (* Push the image as e.g. windows-mingw-20H2-ocaml-4.09  *)
+            let tags = [full_tag] in
+            if switch <> Ocaml_version.Releases.latest then tags
+            else (
+              (* For every distro, also create a link to the latest OCaml compiler.
+                 e.g. windows-mingw-20H2 -> windows-mingw-20H2-ocaml-4.09 *)
+              Tag.v_alias distro :: tags
+            )
+          in
+          (* Fmt.pr "Aliases: %s -> %a@." full_tag Fmt.(Dump.list string) (List.sort String.compare tags); *)
+          let pushes = List.map (fun tag -> Current_docker.push_manifest ?auth:Conf.auth ~tag images |> Current.ignore_value) tags in
+          Some (Switch_map.add switch images Switch_map.empty, (full_tag, Current.all pushes))
+        )
+      end |> List.split
```

Comment:
I haven't stared too hard, but is there not some common code with the linux pipeline above which could be factored out now?

Reply:
There was, but I'm not sure if it improves readability substantially -- might improve maintainance as expected.